src/main/java/team/themoment/hellogsmv3/domain/oneseo/service/ModifyOneseoStatusService.java

```diff
@@ -0,0 +1,62 @@
+package team.themoment.hellogsmv3.domain.oneseo.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import team.themoment.hellogsmv3.domain.member.entity.Member;
+import team.themoment.hellogsmv3.domain.member.service.MemberService;
+import team.themoment.hellogsmv3.domain.oneseo.dto.OneseoStatusReqDto;
+import team.themoment.hellogsmv3.domain.oneseo.entity.EntranceTestResult;
+import team.themoment.hellogsmv3.domain.oneseo.entity.Oneseo;
+import team.themoment.hellogsmv3.domain.oneseo.repository.EntranceTestResultRepository;
+import team.themoment.hellogsmv3.domain.oneseo.repository.OneseoRepository;
+import team.themoment.hellogsmv3.global.exception.error.ExpectedException;
+
+@Service
+@RequiredArgsConstructor
+public class ModifyOneseoStatusService {
+
+    private final OneseoRepository oneseoRepository;
+    private final EntranceTestResultRepository entranceTestResultRepository;
+    private final MemberService memberService;
+
+    public void execute(Long memberId, OneseoStatusReqDto oneseoStatusReqDto) {
+        Member member = memberService.findById(memberId);
+        Oneseo oneseo = oneseoRepository.findByMember(member)
+                .orElseThrow(() -> new ExpectedException("해당 지원자의 원서를 찾을 수 없습니다. member ID : " + memberId, HttpStatus.NOT_FOUND));
+        EntranceTestResult entranceTestResult = entranceTestResultRepository.findByOneseo(oneseo)
+                .orElseThrow(() -> new ExpectedException("해당 지원자의 원서로 입학 시험 결과를 찾을 수 없습니다. member ID : " + oneseo.getMember().getId(), HttpStatus.NOT_FOUND));
+
+        saveModifiedOneseo(oneseo, oneseoStatusReqDto);
+        saveModifiedEntranceTestResult(entranceTestResult, oneseoStatusReqDto);
```

Comment:
```suggestion
        repo.save(buildModifiedOneseo(oneseo, oneseoStatusReqDto));
        repo.save(buildModifiedTestResult(saveModifiedEntranceTestResult(entranceTestResult, oneseoStatusReqDto));
```
개인적인 의견으로는 이 public 서비스에서 DB 로직, `repo.save` 를 노출하는 방향이 더 나을 것 같습니다.
뭔가 구체적인 로직을 private 메소드로 빼서 추상화를 하려고 하신 것 같다는 느낌인데.

그래서 추상화 관점에서 어차피 `save` 라는 걸 노출하고자 하는 목적이시라면, repo.save 로 노출하는것이 더 직관적일 것 같습니다.
부가적으로 SRP 관점에서는 두 가지 이상의 역할(객체 조합 및 DB 저장)을 하는 private 메소드 구성이 된 것 같구요.
마지막으로 해당 클래스에 엔티티 객체를 `build` 하는 로직이 있는게 적절할지도 고민해보시면 좋을 것 같습니다.

Reply:
좋은의견 감사합니다.
고민해보고 반영해보도록하겠습니다!